pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -200,24 +202,31 @@ func (b *blockWithSuccessors) isIncludedIn(other *blockWithSuccessors) bool {
 	return true
 }
 
-// addSuccessorIfPossible adds successor to this block, but only if this block is included in the other block.
-// Successor is not added directly to this block, but correct place in the tree is found for it first.
+// addSuccessorIfPossible adds the given block as a successor of this block, or of a successor of this block
+// Successor is added in the correct place in the tree of successors of this block.
```

Comment:
I think the article is missing:
```suggestion
// The successor is added in the correct place in the tree of successors of this block.
```